### PR TITLE
Unify style of communication with the users for Breeze.

### LINKED ISSUE #22906: Breeze: Describe approach of communicating with users.
We should add custom styles for Breeze (Info/error/warning/guidance/etc.) and add an ADR with rules describing when we should use them.

See discussion in https://github.com/apache/airflow/pull/22901/files#r847173433

### PR DESCRIPTION
**Pull request by potiuk**
Fixes: https://github.com/apache/airflow/issues/22906

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragement file, named `{pr_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Comment by potiuk** [comment]
cc: @blag  - you might be interested to comment (it also adds ADR for that) 

@bowrna - I think you introduced the themes some time ago - and I finally gotten to unify it and explain what is our "approach" for communiction for the future.

**Comment by potiuk** [comment]
Hey here @leahecole  you might be interest - I've added colour-blind-friendly mode to Breeze (speeking of accessibility).

File: BREEZE.rst [hunk 1]
```diff
@@ -378,6 +378,17 @@ default settings.
...
+Another part of configuration is enabling/disabling cheatsheet, asciiart. The cheatsheet and asciiart can
+be disabled - they are "nice looking" and cheatsheet
+contains useful information for first time users but eventually you might want to disable both if you
+find it repetitive and annoying.
+
+With the config setting colour-blind-friendly communication for Breeze messages. By default we communicate
+with the users about information/errors/warnings/successes via colour-coded messages, but we can switch
+it off by passing ``--no-colour`` to config in which case the messages to the user printed by Breeze
+will be printed using different schemes (italic/bold/underline) to indicate different kind of messages
+rather than colours.
```
Comment (eladkal) [inline BREEZE.rst:390]: The whole text is about configurable options.. maybe we can short it by just saying that there are configurable option and have a command to view relevant flags?

**Review by eladkal — APPROVED** [review]
This is great!

**Comment by leahecole** [comment]
> Hey here @leahecole you might be interest - I've added colour-blind-friendly mode to Breeze (speeking of accessibility).

Thank you for tagging me!! I love PRs like this. I don't know a ton about accessibility + CLIs but I went down a rabbit hole looking to see if bold/italic/underline is "best practice" (imho it's a great idea and is better than solely using color but I was curious what other CLIs do, and I'm not an expert so I don't want to be the deciding voice here) - I didn't find anything about that but I did find [this study](https://dl.acm.org/doi/fullHtml/10.1145/3411764.3445544) from the ACM and it reinforces the fact that being able to turn off the ASCII art is an excellent idea 😁

**Comment by potiuk** [comment]
> Thank you for tagging me!! I love PRs like this. I don't know a ton about accessibility + CLIs but I went down a rabbit hole looking to see if bold/italic/underline is "best practice" (imho it's a great idea and is better than solely using color but I was curious what other CLIs do, and I'm not an expert so I don't want to be the deciding voice here) - I didn't find anything about that but I did find [this study](https://dl.acm.org/doi/fullHtml/10.1145/3411764.3445544) from the ACM and it reinforces the fact that being able to turn off the ASCII art is an excellent idea

Oh yeah. I personally like it but I know it's noise for most people :).

Also there is an interesting point which breeze is mostly following (but I have not thought about accessibility):

> Recommendation 5: Ensure that all commands provide status and progress indication.

Especially "success" message is an important one because indeed when the command silently exists you might not realised it did so. I think in vast majority of cases we already have it (and `[success]` style is specifically foreseen for that) but I will pay an extra attention to it :).

**Comment by kaxil** [comment]
> > Hey here @leahecole you might be interest - I've added colour-blind-friendly mode to Breeze (speeking of accessibility).
> 
> Thank you for tagging me!! I love PRs like this. I don't know a ton about accessibility + CLIs but I went down a rabbit hole looking to see if bold/italic/underline is "best practice" (imho it's a great idea and is better than solely using color but I was curious what other CLIs do, and I'm not an expert so I don't want to be the deciding voice here) - I didn't find anything about that but I did find [this study](https://dl.acm.org/doi/fullHtml/10.1145/3411764.3445544) from the ACM and it reinforces the fact that being able to turn off the ASCII art is an excellent idea 😁

Thank for you for pointing to the ACM link @leahecole

**Comment by potiuk** [comment]
cc: @blag ^^ you might be interested to take a look at the comments from @leahecole for Airlfow CLI as well :),

**Pull request merged**